src/main/java/core/basesyntax/FieldXY.java

```diff
@@ -13,7 +13,37 @@
 public class FieldXY {
 
     public void moveRobot(Robot robot, int toX, int toY) {
-        //your code
+        if (robot.getX() < toX) {
+            while (robot.getDirection() != Direction.RIGHT) {
+                robot.turnLeft();
+            }
+            while (robot.getX() < toX) {
+                robot.stepForward();
+            }
```

Comment:
please don't duplicate the code. this `while` makes same work as code at lines 27-29. try to use only one `while` for moving x and same for y

Reply:
Thank you! Fixed this code duplication.